### PR TITLE
public function getNames to expose all marker names

### LINKED ISSUE #93: Get all markers from server
Currently there is no public function member to get all markers from the server. However, one alternative way is to subscribe to `/<node_name>/update_full` topic to get a list of markers name and then do a `server->get(name)` to have access to them which seem to be redundant. A function that can return all the markers would be easier and more useful.

### PR DESCRIPTION
**Pull request by ommmid**
A public function to expose all the names of all markers directly from the server.
Closes #93

### CODE REVIEW

File: src/interactive_marker_server.cpp [hunk 1]
```diff
@@ -407,6 +418,7 @@ void InteractiveMarkerServer::publishInit()
   init_pub_.publish( init );
 }
 
+
```
Comment (jacobperron) [inline src/interactive_marker_server.cpp]: nitpick: revert this unrelated whitespace change.

File: src/interactive_markers/interactive_marker_server.py [hunk 2]
```diff
@@ -382,6 +382,13 @@ def publishInit(self):
 
             self.init_pub.publish(init)
 
+    # return all interactive markers
+    def getNames(self):
+        names = []
+        for marker_context in self.marker_contexts.items():
+            names.append(marker_context.int_marker.name)
+        return names
```
Comment (jacobperron) [inline src/interactive_markers/interactive_marker_server.py]: I suggest we use a list comprehension. Also, I think we should use `keys()` (instead of `items()`) to get the name:

```suggestion
        return [marker_name for marker_name in self.marker_contexts.keys()]
```

File: src/interactive_markers/interactive_marker_server.py [hunk 3]
```diff
@@ -382,6 +382,13 @@ def publishInit(self):
 
             self.init_pub.publish(init)
 
+    # return all interactive markers
```
Comment (jacobperron) [inline src/interactive_markers/interactive_marker_server.py]:
```suggestion
    # return all interactive marker names
```

File: include/interactive_markers/interactive_marker_server.h [hunk 4]
```diff
@@ -156,6 +155,11 @@ class InteractiveMarkerServer : boost::noncopyable
   INTERACTIVE_MARKERS_PUBLIC
   bool get( std::string name, visualization_msgs::InteractiveMarker &int_marker ) const;
 
+  /// Return all interactive markers names in the server
+  INTERACTIVE_MARKERS_PUBLIC
+  /// @param[out] names       Names of all the markers
```
Comment (jacobperron) [inline include/interactive_markers/interactive_marker_server.h]: Please put the docblock _above_ `INTERACTIVE_MARKERS_PUBLIC`.

File: src/interactive_marker_server.cpp [hunk 5]
```diff
@@ -388,6 +388,17 @@ bool InteractiveMarkerServer::get( std::string name, visualization_msgs::Interac
   return false;
 }
 
+void InteractiveMarkerServer::getNames(std::vector<std::string>& names)
```
Comment (jacobperron) [inline src/interactive_marker_server.cpp]: Is there a particular reason you chose to use an output parameter instead of returning a vector? e.g.

```c
std::vector<std::string> InteractiveMarkerServer::getNames()
```

IMO, returning the vector seems more intuitive.

File: src/interactive_marker_server.cpp [hunk 6]
```diff
@@ -388,6 +388,17 @@ bool InteractiveMarkerServer::get( std::string name, visualization_msgs::Interac
...
+{
+  boost::recursive_mutex::scoped_lock lock( mutex_ );
+
+  M_MarkerContext::iterator it;
+  for ( it = marker_contexts_.begin(); it != marker_contexts_.end(); it++ )
+  {
+    names.push_back( it->second.int_marker.name );
```
Comment (jacobperron) [inline src/interactive_marker_server.cpp]: I think this is equivalent to:

```suggestion
    names.push_back( it->first );
```

**Review by jacobperron — COMMENTED** [review]
Thanks for the pull request. I've left some feedback below :)

File: src/interactive_marker_server.cpp (hunk 5 above)
Comment (ommmid) [reply src/interactive_marker_server.cpp]: not a specific reason really, we can return the vector

File: src/interactive_marker_server.cpp (hunk 1 above)
Comment (ommmid) [reply src/interactive_marker_server.cpp]: done

File: include/interactive_markers/interactive_marker_server.h (hunk 4 above)
Comment (ommmid) [reply include/interactive_markers/interactive_marker_server.h]: done

**Review by jacobperron — APPROVED** [review]
